src/main/cpp/jsonrpcprivate.cpp

```diff
@@ -295,9 +295,9 @@ void JsonRpcPrivate::handleResponse(
     if (i != m_callbacks.end()) {
       // Get the shared pointer to the callback
       callback = i->second.lock();
+      // Remove the weak pointer to the callback
+      m_callbacks.erase(i);
     }
-    // Remove the weak pointer to the callback
-    m_callbacks.erase(i);
```

Comment:
Good catch!  I have cherry picked this over to our release candidate branch for now, and will look over the rest of this pull request later.  We switched over to [git flow](http://nvie.com/posts/a-successful-git-branching-model) a while back, so the rest of this will have to go into develop until our next release, and there are a few changes both there and in a feature branch that we are about to merge to develop that will likely conflict with the whitespace parsing changes.